bundles/config/org.eclipse.smarthome.config.discovery.test/src/test/java/org/eclipse/smarthome/config/discovery/internal/AutomaticInboxProcessorTest.java

```diff
@@ -374,10 +380,85 @@ public void testThingWithConfigIsBeingRemoved() {
+        inbox.add(DiscoveryResultBuilder.create(THING_UID3).build());
+        verify(thingRegistry, never()).add(argThat(thing -> thing.getUID().equals(THING_UID3)));
+    }
+
+    @Test
+    public void testAutomaticDiscoveryResultApprovalIfInboxEntriesExistBeforeApprovalPredicatesAreAdded() {
+        inbox.add(DiscoveryResultBuilder.create(THING_UID).build());
+        inbox.add(DiscoveryResultBuilder.create(THING_UID2).build());
+        inbox.add(DiscoveryResultBuilder.create(THING_UID3).build());
+
+        // Adding this inboxAutoApprovePredicate will auto-approve the first two discovery results as they have matching
+        // thing type UID.
+        automaticInboxProcessor.addInboxAutoApprovePredicate(
+                discoveryResult -> discoveryResult.getThingTypeUID().equals(THING_TYPE_UID));
+
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID)));
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID2)));
+        verify(thingRegistry, never()).add(argThat(thing -> thing.getUID().equals(THING_UID3)));
+
+        // Adding this inboxAutoApprovePredicate will auto-approve the third discovery results as it has matching
+        // thing type UID.
+        automaticInboxProcessor.addInboxAutoApprovePredicate(
+                discoveryResult -> discoveryResult.getThingTypeUID().equals(THING_TYPE_UID3));
+
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID)));
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID2)));
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID3)));
+    }
+
+    @Test
+    public void testAlwaysAutoApproveInboxEntries() {
+        // Before setting the always auto approve property, existing inbox results are not approved
+        inbox.add(DiscoveryResultBuilder.create(THING_UID).build());
+
+        verify(thingRegistry, never()).add(argThat(thing -> thing.getUID().equals(THING_UID)));
+
+        // After setting the always auto approve property, all existing inbox results are approved.
+        Map<String, Object> configProperties = new HashMap<>();
+        configProperties.put(AutomaticInboxProcessor.ALWAYS_AUTO_APPROVE_CONFIG_PROPERTY, true);
+        automaticInboxProcessor.activate(configProperties);
+
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID)));
+
+        // Newly added inbox results are also approved.
+        inbox.add(DiscoveryResultBuilder.create(THING_UID2).build());
+        verify(thingRegistry, times(1)).add(argThat(thing -> thing.getUID().equals(THING_UID2)));
+    }
+
+    @Test
+    @Ignore("Should this test pass? It will fail currently, as RuntimeExceptions are not explicitly caught in AutomaticInboxProcessor#isToBeAutoApproved")
```

Comment:
I think its OK for this test to fail. As you already stated before, this `Predicate` is to be implemented by solutions and if they create a failing `Predicate` due to a uncaught `RuntimeException` the framework continues to work. So I think that solutions should take care of these `RuntimeException`s.